src/qdwh.cc

```diff
@@ -0,0 +1,410 @@
+// Copyright (c) 2017-2022, University of Tennessee. All rights reserved.
+// SPDX-License-Identifier: BSD-3-Clause
+// This program is free software: you can redistribute it and/or modify it under
+// the terms of the BSD 3-Clause license. See the accompanying LICENSE file.
+
+#include "slate/slate.hh"
+#include "auxiliary/Debug.hh"
+#include "slate/Matrix.hh"
+#include "internal/internal.hh"
+
+namespace slate {
+
+namespace impl {
+
+//------------------------------------------------------------------------------
+/// Distributed parallel polar decomposition based on QDWH algorithm.
+/// Generic implementation for any target.
+/// @ingroup qdwh_specialization
```

Comment:
Step 2: Replace these docs with the main docs from below.
Step 3: Remove "Target target" from this template.
[more steps below]